Dockerfile.ubi

```
@@ -0,0 +1,237 @@
+ARG CUDA_MAJOR=12
+ARG CUDA_MINOR=8
+ARG CUDA_PATCH=1
+
+FROM nvcr.io/nvidia/cuda:${CUDA_MAJOR}.${CUDA_MINOR}.${CUDA_PATCH}-devel-ubi9
+
+ARG CUDA_MAJOR=12
+ARG CUDA_MINOR=8
+ARG CUDA_PATCH=1
+ARG PYTHON_VERSION
+
+WORKDIR /workspace
+
+ENV LANG=C.UTF-8 \
+    LC_ALL=C.UTF-8 \
+    UV_LINK_MODE=copy \
+    TORCH_CUDA_ARCH_LIST="9.0;10.0+PTX" \
+    PYTHON_VERSION=${PYTHON_VERSION:-3.12} \
+    UV_TORCH_BACKEND=${UV_TORCH_BACKEND:-cu${CUDA_MAJOR}${CUDA_MINOR}} \
+    VIRTUAL_ENV=/opt/vllm 
+
+# Install base packages and EPEL in single layer
+RUN dnf install -y dnf-plugins-core && \
+    dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm && \
+    dnf config-manager --set-enabled epel && \
+    dnf config-manager --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel9/x86_64/cuda-rhel9.repo && \
+    dnf install -y --allowerasing \
+        python${PYTHON_VERSION} python${PYTHON_VERSION}-pip python${PYTHON_VERSION}-wheel \
+        python${PYTHON_VERSION}-devel \
+        # use nvidia's packaged nvshmem
+        nvshmem-cuda-${CUDA_MAJOR} \ 
+        which procps findutils tar \
+        gcc gcc-c++ \
+        make cmake \
+        autoconf automake libtool \
+        git \
+        curl wget \
+        gzip \
+        zlib-devel \
+        openssl-devel \
+        pkg-config \
+        libuuid-devel \
+        glibc-devel \
+        rdma-core-devel \
+        numactl-libs \
+        subunit \
+        pciutils \
+        pciutils-libs \
+        ninja-build \
+    && dnf clean all
+
+# Setup Python virtual environment
+RUN python${PYTHON_VERSION} -m venv /opt/vllm && \
+    ${VIRTUAL_ENV}/bin/pip install --no-cache -U pip wheel uv meson-python ninja pybind11
+
+ENV LIBRARY_PATH="/usr/local/cuda/lib64:/usr/local/lib:/usr/local/lib64" \
+    CPATH="/usr/include:/usr/local/include:/usr/local/cuda/include" \
+    PKG_CONFIG_PATH="/usr/lib/pkgconfig:/usr/local/lib/pkgconfig:/usr/local/lib64/pkgconfig" \
+    TRITON_LIBCUDA_PATH="/usr/lib64" \
+    # Set NVSHMEM paths for CMake discovery
+    NVSHMEM_DIR=/usr/lib64/nvshmem/${CUDA_MAJOR} \
+    PATH="/usr/bin/nvshmem_${CUDA_MAJOR}:${PATH}" \
+    CPATH="/usr/include/nvshmem_${CUDA_MAJOR}:${CPATH}" \
+    LIBRARY_PATH="/usr/lib64/nvshmem/${CUDA_MAJOR}:${LIBRARY_PATH}"
+
+# Build and install gdrcopy
+RUN --mount=type=cache,target=/var/cache/git \
+    git clone https://github.com/NVIDIA/gdrcopy.git && \
+    cd gdrcopy && \
+    PREFIX=/usr/local DESTLIB=/usr/local/lib make lib_install && \
+    cp src/libgdrapi.so.2.* /usr/lib64/ && \
+    ldconfig && \
+    cd .. && rm -rf gdrcopy
+
+# Build and install UCX
+ARG UCX_VERSION="v1.18.1"
+RUN --mount=type=cache,target=/var/cache/git \
+    git clone https://github.com/openucx/ucx.git && \
+    cd ucx && \
+    git checkout ${UCX_VERSION} && \
+    ./autogen.sh && \
+    ./configure \
```

Not sure if we need this, but in my Dockerfile I'm doing this:
```suggestion
    ./contrib/configure-release
```